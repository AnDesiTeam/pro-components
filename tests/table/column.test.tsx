import { mount } from 'enzyme';
import React from 'react';
import { ConfigProvider } from 'antd';
import ProTable from '@ant-design/pro-table';
import { request } from './demo';
import { waitForComponentToPaint } from '../util';
import moment from 'moment';
import { MenuOutlined } from '@ant-design/icons';

describe('Table ColumnSetting', () => {
  it('🎏 render', async () => {
    const callBack = jest.fn();
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: 'Name',
            key: 'name',
            dataIndex: 'name',
            render: (text) => callBack(text),
          },
        ]}
        request={request}
        rowKey="key"
      />,
    );
    await waitForComponentToPaint(html, 1200);
    expect(callBack).toBeCalled();
    expect(callBack).toBeCalledWith('Edward King 0');
  });

  it('🎏 query should parse by valueType', async () => {
    const callBack = jest.fn();
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: 'date',
            key: 'date',
            dataIndex: 'date',
            valueType: 'date',
          },
        ]}
        form={{
          initialValues: {
            date: moment(),
          },
        }}
        request={async (params) => {
          console.log(params);
          callBack(params.date);
          return {
            data: [
              {
                key: '1',
                date: moment(),
              },
            ],
            success: true,
          };
        }}
        rowKey="key"
      />,
    );
    await waitForComponentToPaint(html, 1000);
    expect(callBack).toBeCalled();
    expect(callBack).toBeCalledWith('2016-11-22');
  });

  it('🎏 config provide render', async () => {
    const html = mount(
      <ConfigProvider prefixCls="qixian">
        <ProTable
          size="small"
          columns={[
            {
              title: 'Name',
              key: 'name',
              dataIndex: 'name',
            },
          ]}
          request={request}
          rowKey="key"
        />
      </ConfigProvider>,
    );
    await waitForComponentToPaint(html, 1200);
    expect(html.render()).toMatchSnapshot();
  });

  it('🎏 render text', async () => {
    const callBack = jest.fn();
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: 'Name',
            key: 'name',
            dataIndex: 'name',
            renderText: (text) => callBack(text),
          },
          {
            title: 'Name2',
            key: 'name2',
            dataIndex: 'name2',
            valueType: false,
          },
        ]}
        request={request}
        rowKey="key"
      />,
    );
    await waitForComponentToPaint(html, 1200);
    expect(callBack).toBeCalled();
    expect(callBack).toBeCalledWith('Edward King 0');
  });

  it('🎏 change text by renderText', async () => {
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: 'Name',
            key: 'name',
            dataIndex: 'name',
            renderText: (text) => `${text}2144`,
          },
        ]}
        search={false}
        request={request}
        rowKey="key"
      />,
    );
    await waitForComponentToPaint(html, 1200);
    expect(html.find('td.ant-table-cell')).toMatchSnapshot();
  });
  it('🎏 [dragSort] render drag sort default handle by dragSortKey', async () => {
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: '排序',
            dataIndex: 'sort',
          },
          {
            title: '姓名',
            dataIndex: 'name',
            className: 'drag-visible',
          },
          {
            title: '年龄',
            dataIndex: 'age',
          },
          {
            title: '地址',
            dataIndex: 'address',
          },
        ]}
        request={request}
        rowKey="key"
        dragSortKey="sort"
      />,
    );
    await waitForComponentToPaint(html, 1200);
    expect(html.find('.dragSortDefaultHandle')).toMatchSnapshot();
  });

  it('🎏 [dragSort] render drag sort custom handle by dragSortHandlerRender', async () => {
    const dragHandleRender = (rowData: any, idx: any) => (
      <>
        <MenuOutlined className="dragSortCustomHandle" style={{ cursor: 'grab', color: 'gold' }} />
        {idx + 1} - {rowData.name}
      </>
    );
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: '排序',
            dataIndex: 'sort',
          },
          {
            title: '姓名',
            dataIndex: 'name',
            className: 'drag-visible',
          },
          {
            title: '年龄',
            dataIndex: 'age',
          },
          {
            title: '地址',
            dataIndex: 'address',
          },
        ]}
        request={request}
        rowKey="key"
        dragSortKey="sort"
        dragSortHandlerRender={dragHandleRender}
      />,
    );
    await waitForComponentToPaint(html, 1200);
    expect(html.find('.dragSortCustomHandle')).toMatchSnapshot();
  });
});
