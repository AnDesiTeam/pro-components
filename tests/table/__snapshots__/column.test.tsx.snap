// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table ColumnSetting üéè change text by renderText 1`] = `
Array [
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 02144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 12144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 22144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 32144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 42144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 52144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 62144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 72144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 82144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 92144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 102144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 112144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 122144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 132144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 142144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 152144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 162144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 172144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 182144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 192144
  </td>,
]
`;

exports[`Table ColumnSetting üéè config provide render 1`] = `
<div
  class="qixian-pro-table"
  id="ant-design-pro-table"
>
  <div
    class="qixian-pro-table-search qixian-pro-table-search-query-filter"
  >
    <form
      class="qixian-form qixian-form-horizontal"
    >
      <input
        style="display: none;"
        type="text"
      />
      <div
        class="qixian-row qixian-row-start"
        style="margin-left: -12px; margin-right: -12px;"
      >
        <div
          class="qixian-col qixian-col-8"
          style="padding-left: 12px; padding-right: 12px;"
        >
          <div
            class="qixian-row qixian-form-item"
          >
            <div
              class="qixian-col qixian-form-item-label"
              style="flex: 0 0 80px;"
            >
              <label
                class=""
                for="name"
                title="Name"
              >
                Name
              </label>
            </div>
            <div
              class="qixian-col qixian-form-item-control"
            >
              <div
                class="qixian-form-item-control-input"
              >
                <div
                  class="qixian-form-item-control-input-content"
                >
                  <span
                    class="qixian-input-affix-wrapper"
                    style="width: 100%;"
                  >
                    <input
                      class="qixian-input"
                      id="name"
                      placeholder="ËØ∑ËæìÂÖ•"
                      type="text"
                      value=""
                    />
                    <span
                      class="qixian-input-suffix"
                    >
                      <span
                        aria-label="close-circle"
                        class="anticon anticon-close-circle qixian-input-clear-icon-hidden qixian-input-clear-icon"
                        role="button"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="close-circle"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="qixian-col qixian-col-8 qixian-col-offset-8"
          style="padding-left: 12px; padding-right: 12px; text-align: right;"
        >
          <div
            class="qixian-row qixian-form-item pro-form-query-filter-actions"
          >
            <div
              class="qixian-col qixian-form-item-label"
            >
              <label
                class="qixian-form-item-no-colon"
                title=" "
              >
                 
              </label>
            </div>
            <div
              class="qixian-col qixian-form-item-control"
            >
              <div
                class="qixian-form-item-control-input"
              >
                <div
                  class="qixian-form-item-control-input-content"
                >
                  <div
                    class="qixian-space qixian-space-horizontal qixian-space-align-center"
                  >
                    <div
                      class="qixian-space-item"
                    >
                      <div
                        class="qixian-space qixian-space-horizontal qixian-space-align-center"
                      >
                        <div
                          class="qixian-space-item"
                          style="margin-right: 8px;"
                        >
                          <button
                            class="qixian-btn"
                            type="button"
                          >
                            <span>
                              Èáç ÁΩÆ
                            </span>
                          </button>
                        </div>
                        <div
                          class="qixian-space-item"
                        >
                          <button
                            class="qixian-btn qixian-btn-primary"
                            type="button"
                          >
                            <span>
                              Êü• ËØ¢
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    class="qixian-card"
    style="height: 100%;"
  >
    <div
      class="qixian-card-body"
      style="padding-top: 0px; padding-bottom: 0px;"
    >
      <div
        class="qixian-pro-table-list-toolbar"
      >
        <div
          class="qixian-pro-table-list-toolbar-container"
        >
          <div
            class="qixian-space qixian-space-horizontal qixian-space-align-center qixian-pro-table-list-toolbar-left"
          />
          <div
            class="qixian-space qixian-space-horizontal qixian-space-align-center qixian-pro-table-list-toolbar-right"
          >
            <div
              class="qixian-space-item"
            >
              <div
                class="qixian-space qixian-space-horizontal qixian-space-align-center qixian-pro-table-list-toolbar-setting-items"
              >
                <div
                  class="qixian-space-item"
                  style="margin-right: 12px;"
                >
                  <div
                    class="qixian-pro-table-list-toolbar-setting-item"
                  >
                    <span>
                      <span
                        aria-label="reload"
                        class="anticon anticon-reload"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="reload"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 00-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 01655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 01279 755.2a342.16 342.16 0 01-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 01109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
                <div
                  class="qixian-space-item"
                  style="margin-right: 12px;"
                >
                  <div
                    class="qixian-pro-table-list-toolbar-setting-item"
                  >
                    <span>
                      <span
                        aria-label="column-height"
                        class="anticon anticon-column-height qixian-dropdown-trigger"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="column-height"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M840 836H184c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h656c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8zm0-724H184c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h656c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8zM610.8 378c6 0 9.4-7 5.7-11.7L515.7 238.7a7.14 7.14 0 00-11.3 0L403.6 366.3a7.23 7.23 0 005.7 11.7H476v268h-62.8c-6 0-9.4 7-5.7 11.7l100.8 127.5c2.9 3.7 8.5 3.7 11.3 0l100.8-127.5c3.7-4.7.4-11.7-5.7-11.7H548V378h62.8z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
                <div
                  class="qixian-space-item"
                >
                  <div
                    class="qixian-pro-table-list-toolbar-setting-item"
                  >
                    <span
                      aria-label="setting"
                      class="anticon anticon-setting"
                      role="img"
                      tabindex="-1"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="setting"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="qixian-table-wrapper"
      >
        <div
          class="qixian-spin-nested-loading"
        >
          <div
            class="qixian-spin-container"
          >
            <div
              class="qixian-table qixian-table-small qixian-table-empty"
            >
              <div
                class="qixian-table-container"
              >
                <div
                  class="qixian-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="qixian-table-thead"
                    >
                      <tr>
                        <th
                          class="qixian-table-cell"
                        >
                          Name
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="qixian-table-tbody"
                    >
                      <tr
                        class="qixian-table-placeholder"
                      >
                        <td
                          class="qixian-table-cell"
                        >
                          <div
                            class="qixian-empty qixian-empty-normal"
                          >
                            <div
                              class="qixian-empty-image"
                            >
                              <svg
                                class="qixian-empty-img-simple"
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <g
                                  fill="none"
                                  fill-rule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    class="qixian-empty-img-simple-ellipse"
                                    cx="32"
                                    cy="33"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    class="qixian-empty-img-simple-g"
                                    fill-rule="nonzero"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      class="qixian-empty-img-simple-path"
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <p
                              class="qixian-empty-description"
                            >
                              No Data
                            </p>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
